patchwork/views/series.py

```diff
@@ -20,3 +26,59 @@ def series_mbox(request, series_id):
     )
 
     return response
+
+
+def series_detail(request, project_id, series_id):
+    series = get_object_or_404(Series, id=series_id)
+
+    patches = Patch.objects.filter(series=series)
+
+    context = generic_list(
+        request,
+        series.project,
+        'series-detail',
+        view_args={
+            'project_id': project_id,
+            'series_id': series_id,
+        },
+        patches=patches,
+    )
+
+    context.update({'series': series})
+
+    return render(request, 'patchwork/series-detail.html', context)
+
+
+def series_list(request, project_id):
+    project = get_object_or_404(Project, linkname=project_id)
+    context = {}
+    series_list = (
+        Series.objects.filter(project=project)
+        .only(
+            'submitter',
+            'project',
+            'version',
+            'name',
+            'date',
+            'id',
+        )
+        .select_related('project')
+        .order_by('date')
```

Comment:
Should this not be reversed then to sort it from newest to oldest?
Also are we fetching all of them or are we skipping the archived ones?

Reply:
Agreed. Also, the query needs more work. I'm seeing ~300 queries for listing series (once I remove the lookup for `series-detail` from above and am able to render the page).

![Screen Shot 2024-11-01 at 11 08 32](https://github.com/user-attachments/assets/7b2a9b70-8c5c-4d44-9bd8-463e81f03c44)

This is a problem. Fortunately it should be easy join. If you click on that SQL tab, you'll see a description of the (many) queries:

![Screen Shot 2024-11-01 at 11 08 58](https://github.com/user-attachments/assets/370d6e34-47f1-4f63-a46a-6dd65e847c62)

Clicking on one of these, we can see an example of the kind of issue:

![Screen Shot 2024-11-01 at 11 09 09](https://github.com/user-attachments/assets/3fc82453-b91e-4544-8699-da7aab826304)

That indicates that you need to join on submitter also, which you can do with `select_related`. You also need to join on `patches` and a few others. You should be able to get this down to < 30 queries once you resolve these.